src/screening.f90

```diff
@@ -127,6 +127,65 @@ subroutine calculate_Hilbert_weights(w_disc, w_cont, zeroplus, Hilbert_weights)
     end do
   end subroutine calculate_Hilbert_weights
 
+   subroutine hilbert_transform(Reeps_T, Omegas, spec_eps_T)
+    !! EQ (4), R. Balito et. al. 
+    !!
+    !! Reeps_T Real part of bare polarizability
+    !! Omega Energy of excitation in the electron gas
+    !! spec_eps_T Spectral head of bare polarizability
+    !! 
+    
+    real(r64), allocatable, intent(out) :: Reeps_T(:)
+    real(r64), intent(in) :: Omegas(:)
+    real(r64), intent(in) :: spec_eps_T(:)
+    
+    ! Local variables
+    real(r64) :: in1, in2, fin, bl !, domegas
+    integer(i64) :: nx, kh, np
+
+    nx = size(Omegas)
+    allocate(Reeps_T(nx))
+
+    !domegas = Omegas(2)-Omegas(1)
+    
+    do kh = 1, nx
+        in1=0.0_r64
+        in2=0.0_r64
+        do np=1,nx-1-kh
+            bl = log(real(np+1.0_r64)/real(np))
+            if ( (kh+np).lt.nx) then
```

Comment:
It would be nice to add comments explaining the reasoning for these conditionals. You should do this unless it is absolutely obvious from the context.